fix(curriculum): refactor decimal to binary converter to check for positive numbers only (#54719)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/64007367d54d2a7efbf44fcf.md

Update the second condition in your `if` statement to use the `isNaN()` function to check if the value returned by `parseInt()` is `NaN`.

> Then, add another condition to check if the value returned by `parseInt()` is less then `0`.  This will make sure only positive numbers are accepted by the `checkUserInput` function.

Reviewer: ```suggestion
Then, add another condition to check if the value returned by `parseInt()` is less than `0`.  This will make sure only positive numbers are accepted by the `checkUserInput` function.
```

Reviewer: @jdwilkin4, am i wrong jess regarding the grammartical things ?

# --hints--

You should wrap the value returned by `parseInt()` in the `isNaN()` function.

```js
assert.match(String(checkUserInput), /if\s*\(\s*!\s*numberInput\s*\.\s*value\s*\|\|\s*isNaN\(\s*parseInt\(\s*numberInput\s*\.\s*value\s*\)\s*\)/);
```

> For the condition check `numberInput.value` is less than `0`, wrap `numberInput.value` using the `parseInt()` method.

Reviewer: ```suggestion
Use the `parseInt()` method by wrapping `numberInput.value` to check if `numberInput.value` is less than `0`.
```
